api/src/main/java/edu/cornell/mannlib/vitro/webapp/dynapi/components/N3Template.java

```diff
@@ -34,13 +52,89 @@ public Parameters getProvidedParams() {
 		return new Parameters();
 	}
 
+	public String getN3Text() { return this.n3Text; }
+
 	//endregion
 
 	@Override
 	public OperationResult run(OperationData input) {
-		return null;
+		if (!isInputValid(input)) {
+			return new OperationResult(500);
+		}
+
+		String substitutedN3Template;
+		try {
+			substitutedN3Template=insertParameters(input);
+		}catch (InputMismatchException e){
+			log.error(e);
+			return new OperationResult(500);
+		}
+
+		List<Model> additionModels;
+		try {
+			additionModels = ProcessRdfForm.parseN3ToRDF(Arrays.asList(substitutedN3Template), ProcessRdfForm.N3ParseType.REQUIRED);
+		} catch (Exception e) {
+			log.error("Error while trying to parse N3Template string and create a Jena rdf Model");
+			log.error(e);
+			return new OperationResult(500);
+		}
+
+		AdditionsAndRetractions changes = new AdditionsAndRetractions(additionModels, new ArrayList<Model>());
+		Model writeModel = ModelAccess.on(input.getContext()).getOntModel(templateModel.getName());
+		ProcessRdfForm.applyChangesToWriteModel(changes, null, writeModel,"");
+
+		return new OperationResult(200);
+	}
+
+	private String insertParameters(OperationData input) throws InputMismatchException{
+
+		EditN3GeneratorVTwo gen = new EditN3GeneratorVTwo();
+
+		//Had to convert String to List<String> to be compatible with EditN3GeneratorVTwo methods
+		List<String> n3WithParameters = Arrays.asList(n3Text);
+
+		//region Substitute IRI variables
+		Map<String, List<String>> parametersToUris = requiredParams.params.values().stream()
+				.filter(value->value.getType().getName()=="anyURI")
+				.collect(Collectors.toMap(param -> param.getName(), param -> Arrays.asList(input.get(param.getName()))));
+
+		gen.subInMultiUris(parametersToUris, n3WithParameters);
+		//endregion
+
+		//region Substitute other (literal) variables
+		Map<String, List<Literal>> parametersToLiterals = requiredParams.params.values().stream()
+				.filter(value->value.getType().getName()!="anyURI")
+				.collect(Collectors.toMap(
+						param -> param.getName(),
+						param -> Arrays.asList(ResourceFactory.createTypedLiteral(input.get(param.getName())[0]))));
+
+		gen.subInMultiLiterals(parametersToLiterals, n3WithParameters);
+		//endregion
+
+		//Check if any n3 variables are left without inserted value
+		String[] leftoverVariables = Arrays.stream(n3WithParameters.get(0).split(" "))
+				.filter(token -> token.startsWith("?")).toArray(String[]::new);
+		if(leftoverVariables.length>0){
+			throw new InputMismatchException("N3 template variables:'" + Arrays.toString(leftoverVariables) +
+					"' dont have corresponding input parameters with values that should be inserted in their place.");
+		}
+		return n3WithParameters.get(0);
 	}
 
+	private boolean isInputValid(OperationData input) {
```

Comment:
I have transformed Operation from an interface to an abstract class and move there this method. Here might be the git merge conflict.